### PR TITLE
Do not create `spkg-*.log`, `spkg-*.time` files

### LINKED ISSUE #37887: Installing Sage's python3 spkg produces some untracked files
### Steps To Reproduce

Run `./configure --with-system-python3=no; make python3`. This is with Sage 10.3.beta4 on OS X.

### Expected Behavior

This should succeed, leaving a clean git state.

### Actual Behavior

`git status` returns
```
On branch develop
Your branch is up to date with 'upstream/develop'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	build/pkgs/python3/spkg-legacy-uninstall.log
	build/pkgs/python3/spkg-legacy-uninstall.time

nothing added to commit but untracked files present (use "git add" to track)
```

### Additional Information

Probably the log files should be in `logs/pkgs`, maybe called `python3-spkg-legacy-uninstall.*` in case there could be other `spkg-legacy-uninstall` scripts that might produce log files.

### Environment

```markdown
- **OS**: OS X 14.4.1
- **Sage Version**: 10.3.beta4
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

### PR DESCRIPTION
**Pull request by mkoeppe**
- Clean up after #37391, which used `sage-logger` for prefixing output; generating the log/time files was a side effect, which has not proven to be useful
- Fixes #37887 @jhpalmieri

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

- Depends on #37840 (merged here)

### CODE REVIEW

**Comment by jhpalmieri** [comment]
There is some code duplication. Would it make any sense to have a function that would replace each instance like
```
if [ -f spkg-pipinst ]; then
    sage-logger -P spkg-pipinst "$SAGE_SUDO ./spkg-pipinst"
    if [ $? -ne 0 ]; then
        error_msg "Error running the pipinst script for $PKG_NAME."
        exit 1
    fi
fi
```
by a call to a function defined by something like
```
run_script() {
if [ -f $1 ]; then
    sage-logger -P $1 "$SAGE_SUDO ./$1"
    if [ $? -ne 0 ]; then
        error_msg "Error running the $1 script for $PKG_NAME."
        exit 1
    fi
fi}
```
?

**Comment by mkoeppe** [comment]
I've made it less verbose in a different way, please take a look

**Comment by kwankyu** [comment]
It seems that some of `spkg-*.log` files are generated and some not. Which is which and why?

**Comment by mkoeppe** [comment]
If any are still generated with this PR, that's a bug. Is this what you observed?

File: build/bin/sage-logger [hunk 1]
```diff
@@ -61,15 +68,29 @@ else
...
+    pipestatus=(${PIPESTATUS[*]})
+
+    if [ ${pipestatus[1]} -ne 0 ]; then
+        exit ${pipestatus[1]}
+    else
+        exit ${pipestatus[0]}
+    fi
+fi
+
 timefile="$logdir/$logname.time"
```
Comment (kwankyu) [inline build/bin/sage-logger:87]: Please a comment here saying `# if $logfille is given, then` or something, for readibility.

File: build/bin/sage-spkg [hunk 2]
```diff
@@ -646,40 +618,24 @@ unset V
...
 
 # Run the pre-install script, if any
 if [ -f spkg-preinst ]; then
-    sage-logger -p "$SAGE_SUDO ./spkg-preinst" spkg-preinst.log
-    if [ $? -ne 0 ]; then
-        error_msg "Error running the preinst script for $PKG_NAME."
-        exit 1
-    fi
+    sage-logger -P spkg-preinst "$SAGE_SUDO ./spkg-preinst" spkg-preinst || exit_with_error_msg "Error running the preinst script for $PKG_NAME."
 fi
```
Comment (kwankyu) [inline build/bin/sage-spkg:630]: `spkg-preinst.log` is generated. No?

Comment (mkoeppe) [reply build/bin/sage-spkg:630]: Thanks for catching this; fixed in 7efeba390b3

File: build/bin/sage-logger (hunk 1 above)
Comment (mkoeppe) [reply build/bin/sage-logger:87]: Good idea, done in bfb292ae69a

**Comment by kwankyu** [comment]
Can we test it here, with "CI Fix" tag?

I think there should be some explanation why you introduced the log files and why you think now they are not useful...

**Comment by mkoeppe** [comment]
I don't have a better explanation than what I wrote in the ticket description: "I used sage-logger for prefixing output; generating the log/time files was a side effect, which has not proven to be useful"

**Comment by kwankyu** [comment]
You certainly added the logging facility expecting that the logs would be useful. Then now you simply say "which has not proven to be useful". I am curious what happened in the meanwhile...

**Comment by kwankyu** [comment]
I looked #37391 again. Right, you didn't add the logging feature. As you said, it is a side-effect of using `sage-logger` for prefixing. The generated log files are not useful, and hence here you add a feature to turn them off. 

OK. I misunderstood. Sorry.

**Comment by mkoeppe** [comment]
No worries, these are all the right questions to ask!

**Review by kwankyu — APPROVED** [review]
Working well. LGTM.

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**